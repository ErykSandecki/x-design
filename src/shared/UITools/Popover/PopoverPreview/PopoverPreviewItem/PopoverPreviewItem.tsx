import { FC, ReactNode } from 'react';

// core
import { usePopoverRoot } from '../../PopoverRoot/core/PopoverRootProvider';

// types
import { TPreviewData } from '../../PopoverRoot/core/types';

export type TPopoverPreviewItemProps = {
  children: (activeOption: TPreviewData['activeOption']) => ReactNode;
  id: string;
};

export const PopoverPreviewItem: FC<TPopoverPreviewItemProps> = ({ children, id }) => {
  const { activeOption, previewId } = usePopoverRoot();

  return previewId === id ? children(activeOption) : null;
};

export default PopoverPreviewItem;
