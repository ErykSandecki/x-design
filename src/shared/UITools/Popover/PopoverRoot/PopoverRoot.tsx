import { FC, ReactNode } from 'react';

// core
import { PopoverRootProvider } from './core/PopoverRootProvider';

// types
import { TContext } from './core/types';

export type TPopoverRootProps = TContext & {
  children: ReactNode;
};

export const PopoverRoot: FC<TPopoverRootProps> = ({ children, selected, setSelected }) => (
  <PopoverRootProvider selected={selected} setSelected={setSelected}>
    {children}
  </PopoverRootProvider>
);

export default PopoverRoot;
