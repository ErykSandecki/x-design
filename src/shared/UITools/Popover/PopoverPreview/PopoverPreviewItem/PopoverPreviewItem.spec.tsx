// components
import PopoverPreviewItem from './PopoverPreviewItem';
import PopoverRoot from '../../PopoverRoot/PopoverRoot';

// utils
import { customRender } from 'test';

const mockCallBack = jest.fn();

describe('PopoverPreviewItem snapshots', () => {
  it('should render PopoverPreviewItem', () => {
    // before
    const { asFragment } = customRender(
      <PopoverRoot activeOption="activeOption" previewId="id" selected={false} setSelected={mockCallBack}>
        <PopoverPreviewItem id="id">{(activeOption) => activeOption}</PopoverPreviewItem>
      </PopoverRoot>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render when id is different', () => {
    // before
    const { asFragment } = customRender(
      <PopoverRoot previewId="id" selected={false} setSelected={mockCallBack}>
        <PopoverPreviewItem id="">{(activeOption) => activeOption}</PopoverPreviewItem>
      </PopoverRoot>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
