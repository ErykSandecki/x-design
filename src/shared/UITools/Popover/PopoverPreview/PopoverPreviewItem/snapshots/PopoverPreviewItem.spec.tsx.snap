// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverPreviewItem snapshots should not render when id is different 1`] = `<DocumentFragment />`;

exports[`PopoverPreviewItem snapshots should render PopoverPreviewItem 1`] = `
<DocumentFragment>
  activeOption
</DocumentFragment>
`;
